src/main/kotlin/io/arrowkt/repository/AccountRepository.kt

```diff
@@ -0,0 +1,27 @@
+package io.arrowkt.repository
+
+import io.arrowkt.ErrorOr
+import arrow.core.*
+import arrow.data.NonEmptyList
+import arrow.effects.IO
+import io.arrowkt.model.Account
+import io.arrowkt.model.Balance
+import java.time.LocalDate
+
+interface AccountRepository {
+    fun query(no: String): IO<ErrorOr<Option<Account>>>
+    fun store(a: Account): IO<ErrorOr<Account>>
+    fun query(openedOn: LocalDate): IO<ErrorOr<Sequence<Account>>>
+    fun all(): IO<ErrorOr<Sequence<Account>>>
+
+    fun balance(no: String): IO<ErrorOr<Balance>> = query(no)
+        .map {
+            when (it) {
```

Comment:
is there a better approach then the nested `when`?

Reply:
Not much better but `map {  }.fold({ it }, { it })` works :D

Reply:
Actually do send a PR to Arrow for Either

`fun <A> EitherOf<A, A>.get() = fix().fold({ it }, { it })`

Reply:
I'm not sure how that relates to `EitherOf<A, A>`.... 

here `ErrorOr` is`typealias ErrorOr<A> = Either<NonEmptyList<String>, A>`

Reply:
I thought the right side was a NEL too. Nevermind :D